lib/src/code_generator/type.dart

```diff
@@ -54,8 +54,29 @@ abstract class Type {
   /// Returns whether the dart type and C type string are same.
   bool get sameDartAndCType => sameFfiDartAndCType;
 
-  /// Returns the string representation of the Type, for debugging purposes
-  /// only. This string should not be printed as generated code.
+  /// Returns generated Dart code that converts the given value from its
+  /// DartType to its FfiDartType.
+  String convertDartTypeToFfiDartType(
+    Writer w,
+    String value, {
+    bool objCShouldRetain = false,
```

Comment:
`objCRetain` (It's not a should, it's guaranteed, we generate a `retain: true`.)

Also, let's make it non-optional. I think it would be useful that all call sites explicitly reason about memory management in the generated code.

Reply:
Done